modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java

```diff
@@ -494,6 +484,143 @@ private long _getPublishedLayoutClassPK(long classPK) {
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority()));
+
+		Collections.reverse(aboveSegmentsExperiences);
+
+		List<SegmentsExperience> belowSegmentsExperiences =
+			segmentsExperiencePersistence.findByG_C_C_LtP(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority());
+
+		return Stream.concat(
+			Stream.concat(
+				aboveSegmentsExperiences.stream(),
+				Stream.of(segmentsExperience)),
+			belowSegmentsExperiences.stream()
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private SegmentsExperience _swapAllListSegmentsExperience(
+		int lowestPriority, int priority,
+		SegmentsExperience segmentsExperience) {
+
+		List<SegmentsExperience> segmentsExperiences = _getSegmentExperiences(
+			segmentsExperience);
+
+		if (ListUtil.isEmpty(segmentsExperiences)) {
+			segmentsExperience.setPriority(priority);
+
+			return segmentsExperiencePersistence.update(segmentsExperience);
+		}
+
+		Stream<SegmentsExperience> stream = segmentsExperiences.stream();
+
+		int count = Math.toIntExact(
+			stream.filter(
+				s -> s.getPriority() >= 0
+			).count());
+
+		segmentsExperiences = _updateSegmentExperiencesPriority(
+			segmentsExperiences, lowestPriority - 1, -1);
+
+		segmentsExperiencePersistence.flush();
+
+		_updateSegmentExperiencesPriority(segmentsExperiences, count, -1);
+
+		return segmentsExperiencePersistence.fetchByPrimaryKey(
+			segmentsExperience.getSegmentsExperienceId());
+	}
+
+	private SegmentsExperience _swapSegmentsExperience(
+		int lowestPriority, int priority,
+		SegmentsExperience segmentsExperience) {
+
+		SegmentsExperience swapSegmentsExperience =
+			segmentsExperiencePersistence.fetchByG_C_C_P(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(), priority);
+
+		if (swapSegmentsExperience == null) {
+			segmentsExperience.setPriority(priority);
+
+			return segmentsExperiencePersistence.update(segmentsExperience);
+		}
+
+		int originalPriority = segmentsExperience.getPriority();
+
+		segmentsExperience.setPriority(lowestPriority - 1);
+
+		segmentsExperience = segmentsExperiencePersistence.update(
+			segmentsExperience);
+
+		swapSegmentsExperience.setPriority(lowestPriority - 2);
+
+		swapSegmentsExperience = segmentsExperiencePersistence.update(
+			swapSegmentsExperience);
+
+		segmentsExperiencePersistence.flush();
+
+		segmentsExperience.setPriority(priority);
+
+		segmentsExperience =
+			segmentsExperienceLocalService.updateSegmentsExperience(
+				segmentsExperience);
+
+		swapSegmentsExperience.setPriority(originalPriority);
+
+		segmentsExperienceLocalService.updateSegmentsExperience(
+			swapSegmentsExperience);
+
+		return segmentsExperience;
+	}
+
+	private List<SegmentsExperience> _updateSegmentExperiencesPriority(
+		List<SegmentsExperience> segmentsExperiences, int initialPriority,
+		int counter) {
```

Comment:
Correct if I'm wrong, this property works as an offset? If so, could you call it offset?

Reply:
Awesome feedback :)